### ISSUE TITLE
Error: gzr/command.rb:208:in `rewrite_color_palette!': undefined method `[]' for nil:NilClass

### ISSUE THREAD

**Issue by dsoegijono** [looker-open-source/gzr#79]
I thought it was the same error as #78 but the error message is different. 

```
$ gzr dashboard cat 1
Traceback (most recent call last):
	45: from /Users/daphne/.rvm/gems/ruby-2.5.3/bin/ruby_executable_hooks:24:in `<main>'
	44: from /Users/daphne/.rvm/gems/ruby-2.5.3/bin/ruby_executable_hooks:24:in `eval'
	43: from /Users/daphne/.rvm/gems/ruby-2.5.3/bin/gzr:23:in `<main>'
	42: from /Users/daphne/.rvm/gems/ruby-2.5.3/bin/gzr:23:in `load'
	41: from /Users/daphne/.rvm/gems/ruby-2.5.3/gems/gazer-0.2.42/exe/gzr:36:in `<top (required)>'
	40: from /Users/daphne/.rvm/gems/ruby-2.5.3/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
	39: from /Users/daphne/.rvm/gems/ruby-2.5.3/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	38: from /Users/daphne/.rvm/gems/ruby-2.5.3/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	37: from /Users/daphne/.rvm/gems/ruby-2.5.3/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	36: from /Users/daphne/.rvm/gems/ruby-2.5.3/gems/thor-0.20.3/lib/thor.rb:238:in `block in subcommand'
	35: from /Users/daphne/.rvm/gems/ruby-2.5.3/gems/thor-0.20.3/lib/thor/invocation.rb:115:in `invoke'
	34: from /Users/daphne/.rvm/gems/ruby-2.5.3/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	33: from /Users/daphne/.rvm/gems/ruby-2.5.3/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	32: from /Users/daphne/.rvm/gems/ruby-2.5.3/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	31: from /Users/daphne/.rvm/gems/ruby-2.5.3/gems/gazer-0.2.42/lib/gzr/commands/dashboard.rb:60:in `cat'
	30: from /Users/daphne/.rvm/gems/ruby-2.5.3/gems/gazer-0.2.42/lib/gzr/commands/dashboard/cat.rb:45:in `execute'
	29: from /Users/daphne/.rvm/gems/ruby-2.5.3/gems/gazer-0.2.42/lib/gzr/modules/session.rb:207:in `with_session'
	28: from /Users/daphne/.rvm/gems/ruby-2.5.3/gems/gazer-0.2.42/lib/gzr/commands/dashboard/cat.rb:47:in `block in execute'
	27: from /Users/daphne/.rvm/gems/ruby-2.5.3/gems/gazer-0.2.42/lib/gzr/commands/dashboard/cat.rb:47:in `each_index'
	26: from /Users/daphne/.rvm/gems/ruby-2.5.3/gems/gazer-0.2.42/lib/gzr/commands/dashboard/cat.rb:49:in `block (2 levels) in execute'
	25: from /Users/daphne/.rvm/gems/ruby-2.5.3/gems/gazer-0.2.42/lib/gzr/command.rb:173:in `find_vis_config_reference'
	24: from /Users/daphne/.rvm/gems/ruby-2.5.3/gems/gazer-0.2.42/lib/gzr/command.rb:173:in `each'
	23: from /Users/daphne/.rvm/gems/ruby-2.5.3/gems/gazer-0.2.42/lib/gzr/command.rb:173:in `block in find_vis_config_reference'
	22: from /Users/daphne/.rvm/gems/ruby-2.5.3/gems/gazer-0.2.42/lib/gzr/command.rb:173:in `find_vis_config_reference'
	21: from /Users/daphne/.rvm/gems/ruby-2.5.3/gems/gazer-0.2.42/lib/gzr/command.rb:173:in `each'
	20: from /Users/daphne/.rvm/gems/ruby-2.5.3/gems/gazer-0.2.42/lib/gzr/command.rb:173:in `block in find_vis_config_reference'
	19: from /Users/daphne/.rvm/gems/ruby-2.5.3/gems/gazer-0.2.42/lib/gzr/command.rb:171:in `find_vis_config_reference'
	18: from /Users/daphne/.rvm/gems/ruby-2.5.3/gems/gazer-0.2.42/lib/gzr/commands/dashboard/cat.rb:50:in `block (3 levels) in execute'
	17: from /Users/daphne/.rvm/gems/ruby-2.5.3/gems/gazer-0.2.42/lib/gzr/command.rb:190:in `find_color_palette_reference'
	16: from /Users/daphne/.rvm/gems/ruby-2.5.3/gems/gazer-0.2.42/lib/gzr/command.rb:190:in `each'
	15: from /Users/daphne/.rvm/gems/ruby-2.5.3/gems/gazer-0.2.42/lib/gzr/command.rb:190:in `block in find_color_palette_reference'
	14: from /Users/daphne/.rvm/gems/ruby-2.5.3/gems/gazer-0.2.42/lib/gzr/command.rb:190:in `find_color_palette_reference'
	13: from /Users/daphne/.rvm/gems/ruby-2.5.3/gems/gazer-0.2.42/lib/gzr/command.rb:190:in `each'
	12: from /Users/daphne/.rvm/gems/ruby-2.5.3/gems/gazer-0.2.42/lib/gzr/command.rb:190:in `block in find_color_palette_reference'
	11: from /Users/daphne/.rvm/gems/ruby-2.5.3/gems/gazer-0.2.42/lib/gzr/command.rb:190:in `find_color_palette_reference'
	10: from /Users/daphne/.rvm/gems/ruby-2.5.3/gems/gazer-0.2.42/lib/gzr/command.rb:190:in `each'
	 9: from /Users/daphne/.rvm/gems/ruby-2.5.3/gems/gazer-0.2.42/lib/gzr/command.rb:190:in `block in find_color_palette_reference'
	 8: from /Users/daphne/.rvm/gems/ruby-2.5.3/gems/gazer-0.2.42/lib/gzr/command.rb:190:in `find_color_palette_reference'
	 7: from /Users/daphne/.rvm/gems/ruby-2.5.3/gems/gazer-0.2.42/lib/gzr/command.rb:190:in `each'
	 6: from /Users/daphne/.rvm/gems/ruby-2.5.3/gems/gazer-0.2.42/lib/gzr/command.rb:190:in `block in find_color_palette_reference'
	 5: from /Users/daphne/.rvm/gems/ruby-2.5.3/gems/gazer-0.2.42/lib/gzr/command.rb:190:in `find_color_palette_reference'
	 4: from /Users/daphne/.rvm/gems/ruby-2.5.3/gems/gazer-0.2.42/lib/gzr/command.rb:190:in `each'
	 3: from /Users/daphne/.rvm/gems/ruby-2.5.3/gems/gazer-0.2.42/lib/gzr/command.rb:190:in `block in find_color_palette_reference'
	 2: from /Users/daphne/.rvm/gems/ruby-2.5.3/gems/gazer-0.2.42/lib/gzr/command.rb:188:in `find_color_palette_reference'
	 1: from /Users/daphne/.rvm/gems/ruby-2.5.3/gems/gazer-0.2.42/lib/gzr/commands/dashboard/cat.rb:51:in `block (4 levels) in execute'
/Users/daphne/.rvm/gems/ruby-2.5.3/gems/gazer-0.2.42/lib/gzr/command.rb:208:in `rewrite_color_palette!': undefined method `[]' for nil:NilClass (NoMethodError)
```

**Comment by JackDavidson** [comment]
I'm getting the same error:

```
0.2.42/lib/gzr/command.rb:208:in `rewrite_color_palette!': undefined method `[]' for nil:NilClass (NoMethodError)
	from /var/lib/gems/2.5.0/gems/gazer-0.2.42/lib/gzr/commands/space/export.rb:115:in `block (4 levels) in process_space'
	from /var/lib/gems/2.5.0/gems/gazer-0.2.42/lib/gzr/command.rb:188:in `find_color_palette_reference'
	from /var/lib/gems/2.5.0/gems/gazer-0.2.42/lib/gzr/command.rb:190:in `block in find_color_palette_reference'
	from /var/lib/gems/2.5.0/gems/gazer-0.2.42/lib/gzr/command.rb:190:in `each'
	from /var/lib/gems/2.5.0/gems/gazer-0.2.42/lib/gzr/command.rb:190:in `find_color_palette_reference'
	from /var/lib/gems/2.5.0/gems/gazer-0.2.42/lib/gzr/command.rb:190:in `block in find_color_palette_reference'
	from /var/lib/gems/2.5.0/gems/gazer-0.2.42/lib/gzr/command.rb:190:in `each'
	from /var/lib/gems/2.5.0/gems/gazer-0.2.42/lib/gzr/command.rb:190:in `find_color_palette_reference'
	from /var/lib/gems/2.5.0/gems/gazer-0.2.42/lib/gzr/commands/space/export.rb:114:in `block (3 levels) in process_space'
	from /var/lib/gems/2.5.0/gems/gazer-0.2.42/lib/gzr/command.rb:171:in `find_vis_config_reference'
	from /var/lib/gems/2.5.0/gems/gazer-0.2.42/lib/gzr/command.rb:173:in `block in find_vis_config_reference'
	from /var/lib/gems/2.5.0/gems/gazer-0.2.42/lib/gzr/command.rb:173:in `each'
	from /var/lib/gems/2.5.0/gems/gazer-0.2.42/lib/gzr/command.rb:173:in `find_vis_config_reference'
	from /var/lib/gems/2.5.0/gems/gazer-0.2.42/lib/gzr/command.rb:173:in `block in find_vis_config_reference'
	from /var/lib/gems/2.5.0/gems/gazer-0.2.42/lib/gzr/command.rb:173:in `each'
	from /var/lib/gems/2.5.0/gems/gazer-0.2.42/lib/gzr/command.rb:173:in `find_vis_config_reference'
	from /var/lib/gems/2.5.0/gems/gazer-0.2.42/lib/gzr/commands/space/export.rb:113:in `block (2 levels) in process_space'
	from /var/lib/gems/2.5.0/gems/gazer-0.2.42/lib/gzr/commands/space/export.rb:111:in `each_index'
	from /var/lib/gems/2.5.0/gems/gazer-0.2.42/lib/gzr/commands/space/export.rb:111:in `block in process_space'
	from /var/lib/gems/2.5.0/gems/gazer-0.2.42/lib/gzr/commands/space/export.rb:109:in `each'
	from /var/lib/gems/2.5.0/gems/gazer-0.2.42/lib/gzr/commands/space/export.rb:109:in `process_space'
	from /var/lib/gems/2.5.0/gems/gazer-0.2.42/lib/gzr/commands/space/export.rb:81:in `block in execute'
	from /var/lib/gems/2.5.0/gems/gazer-0.2.42/lib/gzr/modules/session.rb:207:in `with_session'
	from /var/lib/gems/2.5.0/gems/gazer-0.2.42/lib/gzr/commands/space/export.rb:49:in `execute'
	from /var/lib/gems/2.5.0/gems/gazer-0.2.42/lib/gzr/commands/space.rb:80:in `export'
	from /var/lib/gems/2.5.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	from /var/lib/gems/2.5.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	from /var/lib/gems/2.5.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	from /var/lib/gems/2.5.0/gems/thor-0.20.3/lib/thor/invocation.rb:115:in `invoke'
	from /var/lib/gems/2.5.0/gems/thor-0.20.3/lib/thor.rb:238:in `block in subcommand'
	from /var/lib/gems/2.5.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	from /var/lib/gems/2.5.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	from /var/lib/gems/2.5.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	from /var/lib/gems/2.5.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
	from /var/lib/gems/2.5.0/gems/gazer-0.2.42/exe/gzr:36:in `<top (required)>'
	from /usr/local/bin/gzr:23:in `load'
	from /usr/local/bin/gzr:23:in `<main>'
```

going to pin version to 0.2.41. I'll update my post to let you know if that version works

**Comment by dsoegijono** [comment]
I tried 0.2.41 and it worked with some warnings about color palettes.

And I just saw that they've released 0.2.43 that seems to fix the issue for me, so I'm closing this issue. Thanks!
